### PR TITLE
Make device's id persistent between sessions

### PR DESCRIPTION
**Pull request by dersim-davaod**
Checks:

* [x] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
Resolves: #135

### CODE REVIEW

File: MapboxVision/Services/DeviceInfoProvider.swift [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+    }
 
-    private var interruptionStartTime: Date?
+    lazy var id: String = {
+        let defaults = UserDefaults.standard
 
-    func reset() {
-        id = DeviceInfoProvider.generateID()
-    }
+        if let uuid = defaults.object(forKey: Keys.uniqueDeviceIdKey) as? String {
```
Comment (IodaMikeMapbox) [inline MapboxVision/Services/DeviceInfoProvider.swift]:
```suggestion
        if let uuid = defaults.string(forKey: Keys.uniqueDeviceIdKey) {
```

**Comment by IodaMikeMapbox** [comment]
I thought there is no need to store id in userDefaults, it would be enough to have a ref to it in local var and use this var persistent over period of single app usage, in other words we don't want to make an effort to change it each time and don't want to make an effort to make it persistent all the time

File: MapboxVision/VisionManager/VisionManager.swift [hunk 2]
```diff
@@ -217,28 +216,17 @@ public final class VisionManager: BaseVisionManager {
...
         guard isStoppedForBackground else { return }
         isStoppedForBackground = false
         resume()
     }
 
-    private func stopInterruption() {
-        guard let interruptionStartTime = interruptionStartTime else { return }
-
-        let elapsedTime = Date().timeIntervalSince(interruptionStartTime)
-        if elapsedTime >= Constants.foregroundInterruptionResetThreshold {
```
Comment (IodaMikeMapbox) [inline MapboxVision/VisionManager/VisionManager.swift]: suppose we should remove `Constants.foregroundInterruptionResetThreshold` as well

**Comment by dersim-davaod** [comment]
> I thought there is no need to store id in userDefaults, it would be enough to have a ref to it in local var and use this var persistent over period of single app usage, in other words we don't want to make an effort to change it each time and don't want to make an effort to make it persistent all the time

Probably yes, it's not clear from the description of the issue.

**Comment by dersim-davaod** [comment]
@yunikkk has confirmed UUID should be persistent between sessions.

**Comment by chezzdev** [comment]
Meeting was handled by @deniskoronchik and it was decided to have ids that persist across sessions and launches but reset after app reinstall.

**Pull request merged**